README.md

```diff
@@ -58,4 +72,39 @@ const getJwks = buildGetJwks({
 })
 ```
 
+## Integration Examples
+
+### fastify-jwt
+[fastify-jwt](https://github.com/fastify/fastify-jwt) is a Json Web Token plugin for [Fastify](https://www.fastify.io/).
+
+The following example includes a scenario where you'd like to varify a JWT against a valid JWK on any request to your Fastify server.  Any request with a valid JWT auth token in the header will return a successful response, otherwise will respond with an authentication error.
+
+```javascript
+const Fastify = require('fastify')
+const fjwt = require('fastify-jwt')
+const buildGetJwks = require('get-jwks')
+
+const fastify = Fastify()
+const getJwks = buildGetJwks()
+
+fastify.register(fjwt, {
+  decode: { complete: true },
+  secret: async (request, token, callback) => {
+    const { header: { kid, alg }, payload: { iss } } = token
+    const publicKey = await getJwks.getPublicKey({ kid, domain: iss, alg })
+    callback(null, publicKey)
+  }
+})
```

Comment:
this is the reason why I asked earlier if this function exposed by fastify-jwt supported promises. Matteo is suggesting to avoid mixing async/await with callback based code, but here we have the fastify-jwt function which doesn't seem to support promises, and the getPublisKey function which only supports promises. I don't think we should implement a callback flavor of getPublicKey, we should go forward in terms of javascript features, not backwards. 

Hence, I don't see a way to avoid mixing promises and callback code, therefore this code is good enough for me, with the exception that in order to make the two work properly together you need to wrap the await statement in try/catch.

Alternatively, instead of using async/await you can use plain promise methods and do something like:

```js
getPublicKey(...).then(publicKey => callback(null, publicKey), callback)
```

Reply:
Please do not mix promises and callbacks. __Just use async/await__.

Reply:
I understood this, no worries!

Reply:
give us an approval when you're happy with these changes :)